direct_kinematics_ros_plugins/src/DirectKinematicsROSPlugin.cc

```diff
@@ -0,0 +1,1247 @@
+{
+  // ignition::math::Quaterniond orientation(this->modelState.pose.orientation.x,
+  //                                         this->modelState.pose.orientation.y,
+  //                                         this->modelState.pose.orientation.z,
+  //                                         this->modelState.pose.orientation.w);
+  // ignition::math::Vector3d orientation_euler = orientation.Euler();
+  // gzmsg << "=====================" << std::endl;
+  // gzmsg << orientation_euler << std::endl;
+  // ignition::math::Quaterniond orientation_quat;
+  // orientation_quat.Euler(orientation_euler.X(),
+  //                        orientation_euler.Y(),
+  //                        orientation_euler.Z());
+
+  // gazebo_msgs::SetModelState cmd_msg;
+  // cmd_msg.request.model_state = this->modelState;
+  // cmd_msg.request.model_state.pose.orientation.x = orientation_quat.X();
+  // cmd_msg.request.model_state.pose.orientation.y = orientation_quat.Y();
+  // cmd_msg.request.model_state.pose.orientation.z = orientation_quat.Z();
+  // cmd_msg.request.model_state.pose.orientation.w = orientation_quat.W();
+
+  // ignition::math::Quaterniond orientation(this->modelState.pose.orientation.x,
+  //                                         this->modelState.pose.orientation.y,
+  //                                         this->modelState.pose.orientation.z,
+  //                                         this->modelState.pose.orientation.w);
+  // ignition::math::Vector3d orientation_euler = orientation.Euler();
+  // gzmsg << "=====================" << std::endl;
+  // gzmsg << orientation_euler << std::endl;
+  // gzmsg << "---------------------" << std::endl;
+  // gzmsg << orientation.X() << std::endl;
+  // gzmsg << orientation.Y() << std::endl;
+  // gzmsg << orientation.Z() << std::endl;
+  // gzmsg << orientation.W() << std::endl<< std::endl;
+  // gazebo_msgs::SetModelState cmd_msg;
+  // cmd_msg.request.model_state = this->modelState;
+  // cmd_msg.request.model_state.pose.orientation.x = orientation.X();
+  // cmd_msg.request.model_state.pose.orientation.y = orientation.Y();
+  // cmd_msg.request.model_state.pose.orientation.z = orientation.Z();
+  // cmd_msg.request.model_state.pose.orientation.w = orientation.W();
+
+  // this->commandPublisher["Model"].call(cmd_msg);
+
+  // ---------------------------------------- //
+  // ----------- Previous state ------------- //
+  // ---------------------------------------- //
+  gazebo_msgs::SetModelState cmd_msg;
+  cmd_msg.request.model_state.model_name = this->model->GetName();
+  cmd_msg.request.model_state = this->modelState;
+
+  // --------------------------------------- //
+  // ----------- PITCH COMMAND ------------- //
+  // --------------------------------------- //
+  double target_pitch = this->prev_pitch;
+  int _pitch_cmd_type = _msg->pitch_cmd_type;
+  // 1 : battery position, 2: target once , 3: target servo
+  if (_pitch_cmd_type == 1)  // battery position
```

Comment:
Consider using `switch` here instead of a bunch of `if/else if/else`. I find it easier to read and it really shines in cases like this where you're dispatching on the value of a single variable,

Reply:
:+1: